### ISSUE TITLE
Event propagation

### ISSUE THREAD

**Issue by kuzzmi** [chrisui/react-hotkeys#29]
Hi,

I'm not sure if that just stupid me, or it indeed doesn't work as expected.

I have the following structure:

```
Main
   `-- Panel
           |-- List
           `-- Input
```
Main component is wrapped with HotKeys and the mixin applied. Currently I'm playing with one key - `f`. 
`Main` has only `keyMap` defined, but the handler for `f` is defined in `Input` component. The handler gets called _only_ if I place focus on `Input`. 
The idea is to be able to call handler of the hotkey on `Input`, if the focus is placed on its parent, in this case the parent is `Main`.

The idea is to be able to place focus on the input from any place of the app.

Is this possible, did I miss something, or I should pick another way?

**Comment by coldpour** [comment]
I think your issue is similar to my own and I believe it's caused by the focus is on the body when the app initially renders. It's not possible to wrap the body in a HotKeys component for a global map and I haven't yet found a way to have focus start lower down in the tree on app load so that my handlers will notice events. I didn't really understand how the mixin played into all this so that may be the answer somehow.

**Comment by Chrisui** [comment]
Hey guys I completely forgot until a colleague sent me a snippet of my own code that we allow attaching the `HotKeys` to an alternative `DOMElement` instance. I think this *might* address your issue @coldpour.

For example you could do `<HotKeys attach={window} />`. See reference here: https://github.com/Chrisui/react-hotkeys/blob/master/lib/HotKeys.js#L61

As for your issue @kuzzmi I think you are expecting the reverse propagation of what the event system actually provides. If a child (Input) is focused we can say it's parent (Main) is focused too but we cannot say the same in reverse. Therefore if you want to handle a key event that occurs when `Input` or `Panel` or `Main` is in focus then put the handler on `Main`.

Sorry for the slow response!

**Comment by chrisui** [comment]
Closing since this was opened a while ago! Feel free to re-open if you need further help. :)

**Comment by coldpour** [comment]
@chrisui, sorry it took so long, but I just got around to trying out your suggestion and it didn't work for me. I tried the window, body, document, document.body and all those values in strings as well. No matter what I put in the attach prop, I always had to click on the screen before the handlers would fire. I ended up having to do `document.body.addEventListener('keydown', this.onKeyDown.bind(this));` inside `componentDidMount` to get my handler to fire without clicking first.

**Comment by cuongtavan** [comment]
@chrisui : I try with your suggestion but nothing happens, I always need to focus on element
although try the same with @coldpour

**Comment by michaeljohansen** [comment]
If all you need is to listen after page load (without having to focus on the React HotKeys element by clicking on it), then you can just set the React HotKeys element to be your `document.activeElement`, like so:

```
componentDidMount() {
    this.mainDiv.focus();
}
render() {
    return (
        <HotKeys handlers={...}>
            <div ref={(c) => { this.mainDiv = c; }} />
        </HotKeys>
    )
}
```

**Comment by jeremyckahn** [comment]
For anyone else who comes across global keybinding focus issues, it seems that this has been comprehensively addressed in the [2.0.0 release](https://github.com/greena13/react-hotkeys/releases/tag/v2.0.0) with [the `GlobalHotKeys` component](https://github.com/greena13/react-hotkeys#globalhotkeys-component). I switched to that for my app and all of my global keybinding issues have been solved!
